sympy/functions/special/bessel.py

```diff
@@ -241,12 +241,17 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         except (ValueError, NotImplementedError):
             return self
 
-        if exp.is_positive:
+        if exp.is_positive and nu.is_integer and nu.is_nonnegative:
```

Comment:
Why should `nu` be a nonnegative integer?

Reply:
For `besselj` - `nu` does not necessarily need to be an integer, further the nonnegative case is handled by the existing code already. I have instead added this as a special case now, as I think `nu` being integer would be used quite often, and having an iterative method would speed that up.

For `bessely` - `nu` needs to be an integer, but not necessarily nonnegative, according to the series expansion formula that I have implemented (now appended in the PR description).

Reply:
I am not able to figure out how I can add a test for something like - 
```py
assert N(besselj(0.5, x).series(x, n=4), 5) == \                            
        -0.13298*x**2.5 + 0.79788*x**0.5 + O(x**4)
```
It keeps giving assertion error as the coefficients do not equate. Can you help me in fixing this? I have tried a few different functions like `eq`, `S`, `Float` which didn't help

Reply:
Is something like - `(a - b).subs(x, 2).removeO()` where a is the LHS and b is the RHS alright?

Reply:
I don't think that tests with floats are necessary in this case. They are hard to make right because of rounding errors.

Reply:
The iterative method should work with noninteger values of nu as well (with `gamma(nu + 1)` instead of `factorial(nu)`).